torch/_export/serde/serialize.py

```diff
@@ -2188,26 +2182,38 @@ def deserialize_meta_func(serialized_target: str):
             return target
 
         if nn_module_stack_str := metadata.get("nn_module_stack"):
-            # Originally serialized to "key,orig_path,type_str"
-            def import_nn_module_stack(key, path, ty):
-                return key, (path, ty)
-
-            # Helper function that splits strings by commas except for those
-            # encapsulated by parens, which are valid traces.
-            # TODO: Currently this is needed due to indexing Sequential
-            # layers introducing names in the form "layer.slice(1, None, None)".
-            # If that naming is improved, this fancier splitting can probably be
-            # reverted to a simple split by comma.
-            def metadata_split(metadata):
-                # Remove the parentheses and commas inside them
-                metadata = re.sub(r'\(.*?\)', '', metadata)
-                # Split the string by comma, except for those inside parentheses
-                return re.split(r'(?<!\()\s*,\s*(?!\()', metadata)
-
-            nn_module_stack = dict(
-                import_nn_module_stack(*metadata_split(item))
-                for item in nn_module_stack_str.split(ST_DELIMITER)
-            )
+            # explicitly check the serializer version for nn_module_stack BC
+            if self.schema_version.major == 7 and self.schema_version.minor < SCHEMA_VERSION[1]:
```

Comment:
I think it's bad idea to condition on the schema version in serializer. Instead you should just check whether nn_module_stack field is None.

Reply:
@SherlockNoMad is this the kind of explicit version check you are asking for?

Reply:
@yiming0416 We already have some place to check for major version change: https://fburl.com/code/pfvq1pdy, so you don't need to add them here. 

There's no need to explicitly check for minor version since they are minor :P 
I'll be happy to explain this offline if you want me to :)